reference-implementation/lib/readable-stream-abstract-ops.js

```diff
@@ -159,6 +159,42 @@ export function IsReadableStream(x) {
   return true;
 }
 
+export function PutBackIntoReadableStream(stream, chunk) {
+  if (stream._state === 'closed') {
+    throw new TypeError('stream is closed');
```

Comment:
I'm a bit worried about if not supporting this corner case would be problematic for any use case.

Maybe it happens for the MSE's case, too? All data is read, but we have _remaining data_ we want to put back.